pysindy/optimizers/trapping_sr3.py

```diff
@@ -297,48 +307,95 @@ def __init__(
             self.unbias = False
             self.constraint_order = constraint_order
 
+    def _set_Ptensors(self, r):
+        """Make the projection tensors used for the algorithm."""
+        N = int((r ** 2 + 3 * r) / 2.0)
+
+        # delta_{il}delta_{jk}
+        PL_tensor = np.zeros((r, r, r, N))
+        PL_tensor_unsym = np.zeros((r, r, r, N))
+        for i in range(r):
+            for j in range(r):
+                for k in range(r):
+                    for kk in range(N):
+                        if i == kk and j == k:
+                            PL_tensor_unsym[i, j, k, kk] = 1.0
+
+        # Now symmetrize PL
+        for i in range(r):
+            for j in range(N):
+                PL_tensor[:, :, i, j] = 0.5 * (
+                    PL_tensor_unsym[:, :, i, j] + PL_tensor_unsym[:, :, i, j].T
+                )
+
+        # if j == k, delta_{il}delta_{N-r+j,n}
+        # if j != k, delta_{il}delta_{r+j+k-1,n}
+        PQ_tensor = np.zeros((r, r, r, r, N))
+        for i in range(r):
+            for j in range(r):
+                for k in range(r):
+                    for kk in range(r):
+                        for n in range(N):
+                            if (j == k) and (n == N - r + j) and (i == kk):
+                                PQ_tensor[i, j, k, kk, n] = 1.0
+                            if (j != k) and (n == r + j + k - 1) and (i == kk):
+                                PQ_tensor[i, j, k, kk, n] = 1.0 / 2.0
```

Comment:
FYI versions of python starting with 3 automatically convert (int / int) to float, so you can write the more natural 1 / 2.

Reply:
fixed, and thanks for the FYI. I usually assume the worst, so use floats whenever doing division.